### ISSUE TITLE
Should Rooted::ptr be an UnsafeCell?

### ISSUE THREAD

**Issue by Ms2ger** [servo/rust-mozjs#273]
The point is that the GC can mutate its value behind our backs.

CC @eddyb

**Comment by nox** [comment]
I think it should, but we are entering the realm of things that not even a servo-bindgen hack can cover I fear. Should we revisit the idea of writing more bindings by hand?

**Comment by fitzgen** [comment]
Yes, it most definitely should be an `UnsafeCell`. GC things moving and the collector updating pointers is exactly the kind of thing that rustc can't understand and needs to be told about, AFAICT.
